test/python/test_mapper.py

```diff
@@ -200,6 +199,26 @@ def test_already_mapped(self):
 
         self.assertEqual(sorted(cx_qubits), [[3, 4], [3, 14], [5, 4], [9, 8], [12, 11], [13, 4]])
 
+    @requires_qe_access
+    def test_yzy_zyz_cases(self, QE_TOKEN, QE_URL, hub, group, project):
+        """Test mapper function yzy_to_zyz works in previously failed cases.
+
+        See: https://github.com/QISKit/qiskit-terra/issues/607
+        """
+        qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project)
+        if 'ibmqx4' in qiskit.wrapper.available_backends():
+            backend = 'ibmqx4'
+        elif 'ibmq_20_tokyo' in qiskit.wrapper.available_backends():
+            backend = 'ibmq_20_tokyo'
+        else:
+            raise unittest.SkipTest('Do not have the required backend.')
+        circ1 = load_qasm_string(yzy_zyz_1)
+        qobj1 = qiskit.wrapper.compile(circ1, backend)
+        self.assertIsInstance(qobj1, dict)
+        circ2 = load_qasm_string(yzy_zyz_2)
+        qobj2 = qiskit.wrapper.compile(circ2, backend)
+        self.assertIsInstance(qobj2, dict)
```

Comment:
In addition to testing that qobj is created, is it possible to make sure the correct qobj was created? e.g. check one of the rotation angles against expected.

Reply:
In some sense this is what the `abs_inner` check is doing.  Given an input sequence of rotations, it guarantees that the output sequence gives the same rotation matrix to a tolerance set by the `eps` parameter.  In this sense, we are checking all the time for validity, which is why that condition is in there.

Reply:
Here, we only do `yzy` to `zyz` transformations.  However, the tests check for differences between any valid input Euler angles and the output `zyz` angles, so as to be confident that we are actually doing the right transform regardless of initial basis choice.